ios/RNBLEPrinter.m

```diff
@@ -20,90 +27,49 @@ - (dispatch_queue_t)methodQueue
-{
-    m_printer = nil;
-}
-
-RCT_EXPORT_METHOD(getDeviceList:(RCTResponseSenderBlock)successCallback
-                  fail:(RCTResponseSenderBlock)errorCallback) {
-    @try {
-        !_printerArray ? [NSException raise:@"Null pointer exception" format:@"Must call init function first"] : nil;
-        [[PrinterSDK defaultPrinterSDK] scanPrintersWithCompletion:^(Printer* printer){
-            [_printerArray addObject:printer];
-            NSMutableArray *mapped = [NSMutableArray arrayWithCapacity:[_printerArray count]];
-            [_printerArray enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
-                NSDictionary *dict = @{ @"device_name" : printer.name, @"inner_mac_address" : printer.UUIDString};
-                [mapped addObject:dict];
-            }];
-            NSMutableArray *uniquearray = (NSMutableArray *)[[NSSet setWithArray:mapped] allObjects];;
-            successCallback(@[uniquearray]);
-        }];
-    } @catch (NSException *exception) {
-        errorCallback(@[exception.reason]);
-    }
-}
+        printer_ = [[Epos2Printer alloc] initWithPrinterSeries:1 lang:EPOS2_MODEL_ANK];
+        [printer_ setReceiveEventDelegate:self];
 
-RCT_EXPORT_METHOD(connectPrinter:(NSString *)inner_mac_address
-                  success:(RCTResponseSenderBlock)successCallback
-                  fail:(RCTResponseSenderBlock)errorCallback) {
-    @try {
-        __block BOOL found = NO;
-        __block Printer* selectedPrinter = nil;
-        [_printerArray enumerateObjectsUsingBlock: ^(id obj, NSUInteger idx, BOOL *stop){
-            selectedPrinter = (Printer *)obj;
-            if ([inner_mac_address isEqualToString:(selectedPrinter.UUIDString)]) {
-                found = YES;
-                *stop = YES;
-            }
-        }];
+        NSString* target = [NSString stringWithFormat:@"BT:%@", bdAddress];
+        result = [printer_ connect:target timeout:5000];
 
-        if (found) {
-            [[PrinterSDK defaultPrinterSDK] connectBT:selectedPrinter];
-            [[NSNotificationCenter defaultCenter] postNotificationName:@"BLEPrinterConnected" object:nil];
-            m_printer = selectedPrinter;
-            successCallback(@[[NSString stringWithFormat:@"Connected to printer %@", selectedPrinter.name]]);
-        } else {
-            [NSException raise:@"Invalid connection" format:@"connectPrinter: Can't connect to printer %@", inner_mac_address];
+        if (result != EPOS2_SUCCESS) {
+            [NSException raise:@"Invalid connection" format:@"Can't connect to printer %@", bdAddress];
         }
+        NSData* payload = [NSData dataWithBase64EncodedString:text];
+        [printer_ addCommand:payload];
+        result = [printer_ sendData:5000];
+        if (result != EPOS2_SUCCESS) {
+            [NSException raise:@"Print failed" format:@"Error occurred while printing"];
+        }
+        _successCallback = successCallback;
+        _errorCallback = errorCallback;
```

Comment:
do we need to attach errorCallBack for success case too?

Reply:
At this point it is not success case